test/test_common/utility.h

```diff
@@ -534,6 +540,11 @@ class TestUtility {
     return message;
   }
 
+  // Allows pretty printed test names.
+  static std::string http1ParserImplToString(Http1ParserImpl impl) {
+    return impl == Http1ParserImpl::HttpParser ? "HttpParser" : "BalsaParser";
```

Comment:
nit: could be done in a follow up, I think https://github.com/envoyproxy/envoy/pull/24146/files#diff-f21b0fd69ad347bf91f43b74096401ae3477f7a9a0a2660b3598a96565116b10L75-L84 which implements similar pretty printing is more robust. 

Pragmatically I don't think we'd get additional enums here, but if we did we'd silently clobber it :(.

Reply:
Thanks for the suggestion.  Following up in #24226.